examples/vision/simsiam.py

```diff
@@ -0,0 +1,431 @@
+"""
+Title: Self-supervised contrastive learning with SimSiam
+Author: [Sayak Paul](https://twitter.com/RisingSayak)
+Date created: 2021/03/19
+Last modified: 2021/03/20
+Description: Implementation of a self-supervised learning method for computer vision.
+"""
+"""
+Self-supervised learning (SSL) is an interesting branch of study in the field of
+representation learning. SSL systems try to formulate a supervised signal from a corpus
+of unlabeled data points.  An example is we train a deep neural network to predict the
+next word from a given set of words. In literature, these tasks are known as *pretext
+tasks* or *auxiliary tasks*. If we [train such a network](https://arxiv.org/abs/1801.06146) on a huge dataset (such as
+the [Wikipedia text corpus](https://www.corpusdata.org/wikipedia.asp)) it learns very effective
+representations that transfer well to downstream tasks. Language models like
+[BERT](https://arxiv.org/abs/1810.04805), [GPT-3](https://arxiv.org/abs/2005.14165),
+[ELMo](https://allennlp.org/elmo) all benefit from this.
+
+Much like the language models we can train computer vision models using similar
+approaches. To make things work in computer vision, we need to formulate the learning
+tasks such that the underlying model (a deep neural network) is able to make sense of the
+semantic information present in vision data. One such task is to a model to _contrast_
+between two different versions of the same image. The hope is that in this way the model
+will have learn representations where the similar images are grouped as together possible
+while the dissimilar images are further away.
```

Comment:
Suggestion:

- Refactor (repetition of "computer vision")
- Shorten a few sentences (be mindful of the users attention span) without affecting the quality

```suggestion
We can also train computer vision models using similar SSL approaches. The model
(a deep neural network) needs to make sense of the semantic information present
in vision data, and one way of doing that is modeling to _contrast_ between two
different versions of the same image. This way, the model can attempt to learn
representations where similar images are grouped as "together" as possible,
while dissimilar images aren't.
```